src/main/kotlin/io/github/gradlenexus/publishplugin/AbstractTransitionNexusStagingRepositoryTask.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import io.github.gradlenexus.publishplugin.internal.BasicActionRetrier
+import io.github.gradlenexus.publishplugin.internal.NexusClient
+import io.github.gradlenexus.publishplugin.internal.StagingRepository
+import io.github.gradlenexus.publishplugin.internal.StagingRepositoryTransitioner
+import io.github.gradlenexus.publishplugin.internal.StagingRepositoryDescriptorRegistry
+import org.gradle.api.Action
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.options.Option
+import org.gradle.kotlin.dsl.property
+
+@Suppress("UnstableApiUsage")
+abstract class AbstractTransitionNexusStagingRepositoryTask(objects: ObjectFactory, extension: NexusPublishExtension, repository: NexusRepository, registry: Provider<StagingRepositoryDescriptorRegistry>) :
+        AbstractNexusStagingRepositoryTask(objects, extension, repository) {
+
+    @Input
+    val stagingRepositoryId = objects.property<String>().apply {
+        set(registry.map {
+            it[repository.name].stagingRepositoryId
+        })
+    }
+
+    @Option(option = "staging-repository-id", description = "staging repository id to close")
```

Comment:
`close` does seem to be copied from the close task. `staging repository id to transition` would be best from the domain point of view. However, the concept of repo transitioning is usually completely now known by the developers. Therefore, maybe `staging repository id to close/release` or `staging repository id to close or release`?

Reply:
Good catch! How about "staging repository id to transition"?

Reply:
If you think it will be understandable by users, I'm fine with it.

Alternatively, `staging repository id to transition (close or release)`.

Reply:
I pushed the option down to the concrete task classes now so it can be more specific.